format/Message.fbs

```diff
@@ -99,6 +99,14 @@ table RecordBatch {
 
   /// Optional compression of the message body
   compression: BodyCompression;
+
+  /// Some types such as Utf8View are represented using a variable number of buffers.
+  /// For each such Field in the pre-ordered flattened logical schema, there will be
```

Comment:
Just want to make sure I'm understand, for:

```
struct<i:int32, bv:BinaryView, uv: Utf8View>
```

If bv has 2 "Variadic Buffer"s, uv has 1 "Variadic Buffer", how would be `variadicBufferCounts` be? `[2, 1]` or `[0, 2, 1]`?

Reply:
It would be `[2, 1]` since it is only relevant to the subset of fields which carry a variable number of buffers.

Reply:
I agree it is not strictly clear from reading

Reply:
Perhaps the IPC format addition should be covered somewhere in 
https://arrow.apache.org/docs/format/Columnar.html#recordbatch-message?

Reply:
I've added a section describing variadicBufferCounts with an example of explicit flattening